wandb/sdk/internal/tb_watcher.py

```diff
@@ -462,6 +469,18 @@ def _flush(self) -> None:
             )
             print("\t" + ("\n\t".join(dropped_keys)))
         self._data["_step"] = self._step
+        # if not a live run, set the start time to first tf_event's walltime
+        # self._start_time is only 0 when run_proto is initialized by wandb sync
+        if self._start_time == 0:
+            self._start_time = int(self._data.get("_timestamp", None))
+        # use time.time() if it's a live run to avoid non monotonic reporting due to other wandb.log calls
+        # this is a bit of a hack, it will lead to somewhat incorrect time reporting but will ensure
+        # monotonic reporting of runtime and timestamps
+        if self._live_run:
+            self._data["_timestamp"] = time.time()
```

Comment:
Why are these two paths for initializing start_time needed? Why can't we set `self.start_time to the first timestamp we get from the tfevent file.` For both live and synced runs?